src/v/cluster/archival_metadata_stm.h

```diff
@@ -332,7 +332,11 @@ class archival_metadata_stm final : public raft::persisted_stm<> {
     model::offset _last_dirty_at;
 
     // The last replication future
-    std::optional<ss::future<result<raft::replicate_result>>> _last_replicate;
+    struct last_replicate {
+        model::term_id term;
+        ss::shared_future<result<raft::replicate_result>> result;
```

Comment:
nit: could you add a comment explaining the requirement of shared_future here?

Reply:
Will do in a subsequent PR. Merging to get it before the next CDT run.